loader/loader.c

```diff
@@ -1342,6 +1342,24 @@ struct loader_icd_term *loader_icd_add(struct loader_instance *ptr_inst, const s
 
     return icd_term;
 }
+// Closes the library handle in the scanned ICD, free the lib_name string, and zeros out all data
+void loader_unload_scanned_icd(struct loader_instance *inst, struct loader_scanned_icd *scanned_icd) {
+    if (NULL == scanned_icd) {
+        return;
+    }
+    if (scanned_icd->handle) {
+        loader_platform_close_library(scanned_icd->handle);
+        scanned_icd->handle = NULL;
+    }
+    loader_instance_heap_free(inst, scanned_icd->lib_name);
+    scanned_icd->lib_name = NULL;
+    scanned_icd->CreateInstance = NULL;
+    scanned_icd->EnumerateInstanceExtensionProperties = NULL;
+    scanned_icd->GetInstanceProcAddr = NULL;
+    scanned_icd->GetPhysicalDeviceProcAddr = NULL;
+    scanned_icd->api_version = 0;
+    scanned_icd->interface_version = 0;
+}
```

Comment:
I'm not familiar with the loader's code style (this is the first MR that I've read), but `memset` is more concise and more future-proof against new members getting added to the struct. And `memset` would have no additional overhead here (at least for gcc/clang) because the compiler optimizes away the function call for small structs, and essentially compiles to the same code.

Proof that `memset` is an improvement is that this MR forgets to reset `loader_scanned_icd::EnumerateAdapterPhysicalDevices`.

Reply:
Found a couple more places where this should have been done. Thanks again for the extra pair of eyes!